lib/Onyx.js

```diff
@@ -1044,11 +1045,21 @@ function applyMerge(existingValue, changes) {
         return lastChange;
     }
 
-    if (_.isObject(existingValue) || _.every(changes, _.isObject)) {
+    if (_.isObject(existingValue) || _.isObject(lastChange)) {
+        // We want to merge multiple object delates together
+        // If all of the changes are objects, we can simply merge all of them together
+        // If the batched changes contain nullish values or arrays, we should only merge the very last objects in the batch
```

Comment:
Can you please expand this comment to explain _why_ only the last objects in the batch should be merged? I'm not sure I understand the scenario or what nullish and array values have to do with it.

Reply:
+1
May be we want to merge the very last value in the batch - irrespective of if it is null or not null